framework/src/interfaces/Coupleable.C

```diff
@@ -224,7 +224,12 @@ Coupleable::coupled(const std::string & var_name, unsigned int comp)
   checkVar(var_name);
 
   if (!isCoupled(var_name))
+  {
+    // make sure we don't try to access default var ids that were not provided
+    if (comp + 1 > _optional_var_index[var_name].size())
+      mooseError("Requested component ", comp, " of coupled value ", var_name, " is out of range.");
```

Comment:
Improve the error message like this: `mooseError(_c_name, <original error message>`. This will tell people which object has the problem.